@jkelling weird, I can see the log. Here is the relevant warning treated as error:
`D:\a\alpaka\alpaka\include\alpaka/block/shared/st/BlockSharedMemStMember.hpp(57,1): warning C4267: '+=': conversion from 'size_t' to 'unsigned int', possible loss of data [D:\a\alpaka\alpaka\build\test\unit\atomic\atomic.vcxproj]`

--- a/include/alpaka/alpaka.hpp
+++ b/include/alpaka/alpaka.hpp
@@ -46,12 +46,14 @@
         // dynamic
         #include <alpaka/block/shared/dyn/BlockSharedMemDynUniformCudaHipBuiltIn.hpp>
         #include <alpaka/block/shared/dyn/BlockSharedMemDynBoostAlignedAlloc.hpp>
+        #include <alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp>
         #include <alpaka/block/shared/dyn/Traits.hpp>
         //-----------------------------------------------------------------------------
         // static
         #include <alpaka/block/shared/st/BlockSharedMemStUniformCudaHipBuiltIn.hpp>
         #include <alpaka/block/shared/st/BlockSharedMemStMasterSync.hpp>
         #include <alpaka/block/shared/st/BlockSharedMemStNoSync.hpp>
+        #include <alpaka/block/shared/st/BlockSharedMemStMember.hpp>
         #include <alpaka/block/shared/st/Traits.hpp>
     //-----------------------------------------------------------------------------
     // sync

--- a/include/alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp
+++ b/include/alpaka/block/shared/dyn/BlockSharedMemDynMember.hpp
@@ -28,6 +28,10 @@ namespace alpaka
         {
             namespace dyn
             {
+#if BOOST_COMP_MSVC || defined(BOOST_COMP_MSVC_EMULATED)
+    #pragma warning(push)
+    #pragma warning(disable: 4324)  // warning C4324: structure was padded due to alignment specifier
+#endif
                 //#############################################################################
                 //! Dynamic block shared memory provider using fixed-size
                 //! member array to allocate memory on the stack or in shared
@@ -88,6 +92,9 @@ namespace alpaka
                     mutable std::array<uint8_t, staticAllocBytes> m_mem;
                     unsigned int m_dynSize;
                 };
+#if BOOST_COMP_MSVC || defined(BOOST_COMP_MSVC_EMULATED)
+    #pragma warning(pop)
+#endif
 
                 namespace traits
                 {

--- a/include/alpaka/block/shared/st/BlockSharedMemStMember.hpp
+++ b/include/alpaka/block/shared/st/BlockSharedMemStMember.hpp
@@ -86,9 +86,10 @@ namespace alpaka
 #endif
 
                         template<typename T>
-                        static constexpr size_t alignPitch()
+                        static constexpr unsigned int alignPitch()
                         {
-                            return (sizeof(T)/TDataAlignBytes + (sizeof(T)%TDataAlignBytes>0))*TDataAlignBytes;
+                            return (static_cast<unsigned int>(sizeof(T))/TDataAlignBytes
+                                + static_cast<unsigned int>(sizeof(T))%TDataAlignBytes>0)*TDataAlignBytes;
                         }
                     };
                 }

